lib/internal/worker.js

```diff
@@ -415,6 +423,37 @@ function makeResourceLimits(float64arr) {
   };
 }
 
+function eventLoopUtilization(util1, util2) {
+  // Cache loopStart, since it's only written to once.
+  if (this[kLoopStartTime] <= 0) {
+    this[kLoopStartTime] = this[kHandle].loopStartTime();
+    if (this[kLoopStartTime] <= 0)
+      return { idle: 0, active: 0, utilization: 0 };
+  }
+
+  if (util2) {
+    const idle = util1.idle - util2.idle;
+    const active = util1.active - util2.active;
+    return { idle, active, utilization: active / (idle + active) };
+  }
+
+  const idle = this[kHandle].loopIdleTime();
```

Comment:
Keep in mind that `idle` can be `-1` even if `this[kLoopStartTime]` was positive.

Reply:
Whoops. Thanks.